test/functional/test_runner.py

```diff
@@ -574,6 +577,11 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=
     if not result.wasSuccessful():
         sys.exit("Early exiting after failure in TestFramework unit tests")
 
+    # Exit the test if there is not enough space on the testing dir
+    testdir_disk_usage_stats = shutil.disk_usage(tmpdir)
+    if (testdir_disk_usage_stats.free < MIN_FREE_SPACE):
+        sys.exit(f"Tests aborted: Insufficient space available on the device. Directory: {tmpdir}")
```

Comment:
I think we can reword this to 
`Tests aborted: Insufficient space available in directory: {tmpdir}`

Since the device may have space but the `tmpdir` we are using might have insufficient space